{ 
  "text" : "The NFS implementation in Linux kernel before 2.6.31-rc6 calls certain functions without properly initializing certain data, which allows local users to cause a denial of service (NULL pointer dereference and O_DIRECT oops), as demonstrated using diotest4 from LTP.\n" ,
  "entities" : [ { 
    "eid" : "E1" ,
    "keyid" : "Identity#e206a54e97690cce50cc872dd70ee896" ,
    "type" : "Identity" ,
    "subType" : "Organization" ,
    "value" : "linux" ,
    "begin" : 26 ,
    "end" : 31 ,
    "canonicalForm" : "Linux" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E2" ,
    "keyid" : "Product#e206a54e97690cce50cc872dd70ee896" ,
    "type" : "Product" ,
    "subType" : "OperatingSystem" ,
    "value" : "linux" ,
    "begin" : 26 ,
    "end" : 31 ,
    "canonicalForm" : "Linux" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E3" ,
    "keyid" : "48967fd9a1b1949a175ddec2c2bc2d6e" ,
    "type" : "MiscEntity" ,
    "subType" : "user" ,
    "value" : "local user" ,
    "begin" : 138 ,
    "end" : 149 ,
    "canonicalForm" : "local user" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E4" ,
    "keyid" : "TTP#9c98ef6a7451d6f82ca2194ee23d1acb" ,
    "type" : "TTP" ,
    "subType" : "TTP" ,
    "value" : "denial of service" ,
    "begin" : 161 ,
    "end" : 178 ,
    "canonicalForm" : "denial of service" ,
    "score" : "0.0"
  }
  ,  ] ,
  "relations" : [ { 
    "rid" : "R1" ,
    "type" : "relatedEntity" ,
    "predicate" : { } ,
    "arguments" : [ { 
      "argnum" : 1 ,
      "begin" : 26 ,
      "end" : 31 ,
      "eid" : "E1" ,
      "canonicalForm" : "Linux"
    }
    , { 
      "argnum" : 2 ,
      "begin" : 161 ,
      "end" : 178 ,
      "eid" : "E4" ,
      "canonicalForm" : "denial of service"
    }
    ,  ]
  }
  , { 
    "rid" : "R2" ,
    "type" : "relatedEntity" ,
    "predicate" : { } ,
    "arguments" : [ { 
      "argnum" : 1 ,
      "begin" : 26 ,
      "end" : 31 ,
      "eid" : "E2" ,
      "canonicalForm" : "Linux"
    }
    , { 
      "argnum" : 2 ,
      "begin" : 161 ,
      "end" : 178 ,
      "eid" : "E4" ,
      "canonicalForm" : "denial of service"
    }
    ,  ]
  }
  , { 
    "rid" : "R3" ,
    "type" : "cause" ,
    "predicate" : { 
      "begin" : 150 ,
      "end" : 158 ,
      "surfaceForm" : "to cause"
    }
     ,
    "arguments" : [ { 
      "argnum" : 1 ,
      "begin" : 138 ,
      "end" : 149 ,
      "eid" : "E3" ,
      "canonicalForm" : "local user"
    }
    , { 
      "argnum" : 2 ,
      "begin" : 161 ,
      "end" : 178 ,
      "eid" : "E4" ,
      "canonicalForm" : "denial of service"
    }
    ,  ]
  }
  ,  ]
}
